test/DynamoCoreWpfTests/ConnectorViewModelTests.cs

```diff
@@ -0,0 +1,265 @@
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 1);
+
+            ///hard-coded position (2) on wire
+            connectorViewModel.PanelX = 244;
+            connectorViewModel.PanelY = 303.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 2);
+        }
+
+        /// <summary>
+        /// Check to see a pin can be removed from a connector
+        /// </summary>
+        [Test]
+        public void CanUnPinFromConnector()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+
+            ///hard-coded position (1) on wire
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+
+            connectorViewModel.PinConnectorCommand.Execute(null);
+
+            ///Request remove pin
+            var firstPin = connectorViewModel.ConnectorPinViewCollection.First();
+            firstPin.OnRequestRemove(firstPin, System.EventArgs.Empty);
+
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 0);
+        }
+
+        /// <summary>
+        /// Check to see if can select connected nodes.
+        /// </summary>
+        [Test]
+        public void CanSelectConnectedNodes()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+            connectorViewModel.SelectConnectedCommand.Execute(null);
+            ///Should result in 2 selected nodes.
+            Assert.AreEqual(DynamoSelection.Instance.Selection.Count, 2);
+        }
+        /// <summary>
+        /// Check to see if can break connection.
+        /// </summary>
+        [Test]
+        public void CanBreakConnection()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+            connectorViewModel.BreakConnectionCommand.Execute(null);
+            ///This should result in no connectors existing anymore.
+            var newConnectorsAfterBreak = this.ViewModel.CurrentSpaceViewModel.Connectors;
+            Assert.AreEqual(newConnectorsAfterBreak.Count, 0);
+        }
+
+        /// <summary>
+        /// Check to see if can hide connector.
+        /// </summary>
+        [Test]
+        public void CanHideConnector()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+            ///Toggles hide (visibility == off)
+            connectorViewModel.HideConnectorCommand.Execute(null);
+            Assert.IsFalse(connectorViewModel.IsVisible);
+        }
```

Comment:
i thought we just set it to transparent, didnt we?

Reply:
Correct, however `IsVisible` is the property/flag that indicates to the xaml (datatrigger) that this is so.

Reply:
ohhh right, got this confused with the view `IsVisible` property!